### PR TITLE
Snap results

### PR DESCRIPTION
**Pull request by benvanbasten-ns**
In case the slug of an added result doesn't match the slug of the selected grid (and both slugs are non-None), it checks for a grid whose slug does match. If so, it removes the result and readds it to the other grid.

In case no other grid is found, a messagebar appears and the result is removed from the model.

### CODE REVIEW

File: threedi_plugin.py [hunk 1]
```diff
@@ -148,16 +148,14 @@ def initGui(self):
...
         self.model.result_unchecked.connect(self.map_animator.results_changed)
+        iface.mapCanvas().temporalController().updateTemporalRange.connect(self.map_animator.update_results)
 
-        # start animating
-        tc = iface.mapCanvas().temporalController()
-        tc.updateTemporalRange.connect(self.map_animator.update_results)
-
-        self.model.result_added.connect(self.graph_tool.result_added)
+        # mapping graph tool
+        self.validator.result_validated.connect(self.graph_tool.result_added)
```
Comment (benvanbasten-ns) [inline threedi_plugin.py:158]: Note that the graph now responds to validated results

File: threedi_plugin_model_validation.py [hunk 2]
```diff
@@ -55,36 +53,59 @@ def fail(msg):
...
-                    )
-                return fail("Aggregate results file cannot be opened.")
+            # Really wrong grid, find a grid with the right slug, if not available, abort with pop-up
+            root_node = result_item.model().invisibleRootItem()
+            for i in range(root_node.rowCount()):
+                other_grid_item = root_node.child(i)
 
-        item.setIcon(QIcon(":images/themes/default/mIconSuccess.svg"))
-        self.result_validated.emit(item, True)
+                package = ogr.GetDriverByName('GPKG').Open(str(other_grid_item.path), True)
```
Comment (arjanverkerk) [inline threedi_plugin_model_validation.py:94]: Can we save this on the `grid_item` when loading a grid?

**Review by arjanverkerk — COMMENTED** [review]
Can we consider the alternative once more? Because this is doing validation over again. What if we do first validation of a result and only then add it to the model under the matching parent (so we reverse the order, instead of add => validate we do validate => add)?

**Comment by benvanbasten-ns** [comment]
closing due to different approach

**Pull request closed without merging**